x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { InfraClientCoreSetup } from '../types';
+import { LogsFetchDataResponse } from '../../../observability/public';
+
+export function getLogsHasDataFetcher(getStartServices: InfraClientCoreSetup['getStartServices']) {
```

Comment:
@cauemarcondes I understand this function is a requirement for the dashboard. Is it a hard requirement?

I can imagine that the reason to have it is to not perform a potentially expensive query to fetch the data if there's no data to fetch.

if that's the case, how expensive would it be to call the data fetching function when there's no data at all, versus calling two functions every time?

Reply:
@afgomez, the idea behind the `hasData` function is indeed to get a flag indicating if there's any data available in the `logs` indices. This exists because we have two pages in the Observability UI `/landing` and `/overview` plus `/`. The `/` URL, will call the `hasData` from all plugins,  if one of the results is `true` we redirect the user to `/overview` otherwise `/landing`. 

We could, of course, call the `fetchData` instead of `hasData`, but in that case, the response time would probably be twice or even thrice time slower then only checking if there's data available.

Also `hasData` must return false, if a user doesn't have privileges to read from the logs indices.